Gencred expiration date set to seconds instead of days

chaodaiG opened the pull request:
As tested manually, that the largest allowed expiration duration for the generated token is 48 hours, so it doesn't make much sense to set expiration date on the scale of days.

cjwagner commented on gencred/cmd/gencred/main.go:
```diff
@@ -60,7 +61,7 @@ func (o *options) parseFlags() {
 	flag.BoolVarP(&o.certificate, "certificate", "c", false, "Authorize with a client certificate and key.")
 	flag.BoolVarP(&o.serviceaccount, "serviceaccount", "s", false, "Authorize with a service account.")
-	flag.IntVar(&o.duration, "duration", defaultDurationInDays, "How many days the cred is valid, default is 7.")
+	flag.Int64Var(&o.duration, "duration", defaultDurationInDays, "How many seconds the cred is valid, default is 172800, can only set to be lower than the default.")
```
This change is backwards incompatible. The flag was only introduced 20 days ago so it might be ok if we're confident we're the only people using it.

cjwagner replied:
A `flag.DurationVar` would be more readable.

cjwagner commented on gencred/cmd/gencred/main.go:
```diff
@@ -39,7 +39,8 @@ const (
 	defaultConfigFileName = "/dev/stdout"
-	defaultDurationInDays = 7
+	// Two days. This is the largest allowed value for token expiration.
+	defaultDurationInDays = 172800
```
This is the duration in seconds, not days.

chaodaiG commented:
This PR is more like a short term solution. Since the longer term solution is already up for review https://github.com/kubernetes/test-infra/pull/26246, I'll close this one. @cjwagner I have addressed your comment in the new PR, please take a look there.

/close
